src/lib/utils/plugin_manager.hpp

```diff
@@ -26,6 +26,8 @@ class PluginManager : public Noncopyable {
   void load_plugin(const std::filesystem::path& path);
   void unload_plugin(const PluginName& name);
 
+  const std::vector<PluginName> loaded_plugins() const;
```

Comment:
I believe there is hardly any reason to ever return a const value. References yes, but returns by value are by definition the property of the caller.